bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/services/RefreshThermostatsService.java

```diff
@@ -0,0 +1,99 @@
+import org.openhab.binding.ojelectronics.internal.ThermostatHandler;
+import org.openhab.binding.ojelectronics.internal.models.thermostat.ThermostatModel;
+import org.openhab.binding.ojelectronics.internal.models.thermostat.ThermostatModelBase;
+import org.openhab.binding.ojelectronics.internal.models.thermostat.ThermostatRealTimeValuesModel;
+import org.openhab.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Refreshes values of {@link ThermostatHandler}
+ *
+ * @author Christian Kittel - Initial Contribution
+ */
+@NonNullByDefault
+public class RefreshThermostatsService {
+
+    private final ArrayList<ThermostatModel> thermostats;
+    private final Logger logger = Objects.requireNonNull(LoggerFactory.getLogger(RefreshThermostatsService.class));
+    private final List<Thing> things;
+    private final ArrayList<ThermostatRealTimeValuesModel> realTimeValues;
+
+    /**
+     * Creates a new instance of {@link RefreshThermostatsService}
+     *
+     * @param thermostats {@link ThermostatModel}
+     * @param things Things
+     */
+    public RefreshThermostatsService(ArrayList<ThermostatModel> thermostats, List<Thing> things) {
+        this(thermostats, new ArrayList<>(), things);
+    }
+
+    /**
+     * Creates a new instance of {@link RefreshThermostatsService}
+     *
+     * @param thermostats {@link ThermostatModel}
+     * @param realTimeValues {@link ThermostatRealTimeValuesModel}
+     * @param things Things
+     */
+    public RefreshThermostatsService(ArrayList<ThermostatModel> thermostats,
+            ArrayList<ThermostatRealTimeValuesModel> realTimeValues, List<Thing> things) {
+        this.thermostats = thermostats;
+        this.things = things;
+        this.realTimeValues = realTimeValues;
+        if (this.things.isEmpty()) {
+            logger.warn("Bridge contains no thermostats.");
+        }
+    }
+
+    /**
+     * Handles the changes to all things.
+     */
+    public synchronized void handle() {
+        thermostats.forEach(thermostat -> handleThermostat(thermostat, this::handleThermostatRefresh));
+        realTimeValues.forEach(thermostat -> handleThermostat(thermostat, this::handleThermostatRealTimeValueRefresh));
+    }
+
+    private <T extends ThermostatModelBase> void handleThermostat(T thermostat,
+            BiConsumer<ThermostatHandler, T> refreshHandler) {
+        things.stream().filter(thing -> thing.getHandler() instanceof ThermostatHandler)
+                .map(thing -> (ThermostatHandler) thing.getHandler())
+                .filter(thingHandler -> thingHandler.getSerialNumber().equals(thermostat.serialNumber))
+                .forEach(thingHandler -> {
+                    try {
+                        refreshHandler.accept(Objects.requireNonNull(thingHandler), thermostat);
+                    } catch (Exception e) {
+                        logger.error("Error Handling Refresh of thermostat {} {}", thermostat, e);
```

Comment:
Either like that to log the exception stack trace:
`logger.error("Error Handling Refresh of thermostat {}", thermostat, e);`
Or just to log exception message:
`logger.error("Error Handling Refresh of thermostat {} {}", thermostat, e.getMessage());`

Please do not use ERROR level.